docsrc/examples/Using External C++.ipynb

```diff
@@ -0,0 +1,157 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Advanced Topic: Using External C++ Functions\n",
+    "\n",
+    "This is based on the relevant portion of the CmdStan documentation [here](https://mc-stan.org/docs/cmdstan-guide/using-external-cpp-code.html)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Consider the following Stan model, based on the bernoulli example."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {"nbsphinx": "hidden"},
+   "outputs": [],
+   "source": [
+    "import os\n",
+    "try:\n",
+    "    os.remove('bernoulli_external')\n",
+    "except:\n",
+    "    pass"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from cmdstanpy import CmdStanModel\n",
+    "model_external = CmdStanModel(stan_file='bernoulli_external.stan', compile=False)\n",
+    "print(model_external.code())"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "As you can see, it features a function declaration for `make_odds`, but no definition. If we try to compile this, we will get an error. "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "model_external.compile()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Even enabling the `--allow_undefined` flag to stanc3 will not allow this model to be compiled quite yet."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "model_external.compile(stanc_options={'allow_undefined':True})"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "To resolve this, we need to both tell the Stan compiler an undefined function is okay **and** let C++ know what it should be. \n",
+    "\n",
+    "We can provide a definition in a C++ header file by using the `user_header` argument to either the CmdStanModel constructor or the `compile` method. \n",
+    "\n",
+    "This will enables the `allow_undefined` flag automatically."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "model_external.compile(user_header='make_odds.hpp')"
```

Comment:
update this to `cpp_opts`

Reply:
I'm not sure what you mean. The user interface is still to use `model_header`, this just gets turned into part of the CPP options during validation

Reply:
you're right - user error on my part - testing in wrong environment.